endpointmanager/pkg/capabilityhandler/capabilityreceiver.go

```diff
@@ -84,41 +100,26 @@ func saveMsgInDB(message []byte, args *map[string]interface{}) error {
 }
 
 // CapabilityReceiver receives the capability statement from the queue and adds it to the database
-func CapabilityReceiver(store endpointmanager.FHIREndpointStore) {
-	// Get the config information that can then be used below
-	err := config.SetupConfig()
-	failOnError(err)
-
-	// Set up the queue for sending messages
-	qUser := viper.GetString("quser")
-	qPassword := viper.GetString("qpassword")
-	qHost := viper.GetString("qhost")
-	qPort := viper.GetString("qport")
-	qName := viper.GetString("capquery_qname")
-	messageQueue, channelID, err := queue.ConnectToQueue(qUser, qPassword, qHost, qPort, qName)
-	failOnError(err)
-	log.Info("Successfully connected to Queue!")
-	defer messageQueue.Close()
+func CapabilityReceiver(store endpointmanager.FHIREndpointStore,
+	messageQueue lanternmq.MessageQueue,
+	channelID lanternmq.ChannelID,
+	qName string) error {
 
 	args := make(map[string]interface{})
 	args["store"] = store
+	args["ctx"] = context.Background()
```

Comment:
my gut is that ctx should be passed into CapabilityReceiver:

``` 
func CapabilityReceiver(ctx context.Context,
        store endpointmanager.FHIREndpointStore,
	messageQueue lanternmq.MessageQueue,
	channelID lanternmq.ChannelID,
	qName string) error {
```

and then added to `args`:

```
  args["ctx"] = ctx
```

Reply:
👍